app/src/main/kotlin/io/homeassistant/companion/android/settings/gestures/GesturesViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package io.homeassistant.companion.android.settings.gestures
+
+import androidx.compose.runtime.mutableStateMapOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.homeassistant.companion.android.common.data.prefs.PrefsRepository
+import io.homeassistant.companion.android.common.util.GestureAction
+import io.homeassistant.companion.android.common.util.HAGesture
+import javax.inject.Inject
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class GesturesViewModel @Inject constructor(
+    private val prefsRepository: PrefsRepository,
+) : ViewModel() {
+
+    val gestureActions = mutableStateMapOf<HAGesture, GestureAction>()
+
+    init {
+        viewModelScope.launch {
+            HAGesture.entries.forEach {
+                gestureActions[it] = prefsRepository.getGestureAction(it)
+            }
+        }
+    }
```

Comment:
I'm not sure if we want to tackle this here but we should be sure that getGestureAction internal IO call to the storage is made on IO thread. Right now even if it async `viewModelScope` by default use the main thread group.

Reply:
That is out of scope for this PR in my opinion and should be tackled in the LocalStorage interface then.